packages/common-scripts/src/p2pkh.ts

```diff
@@ -0,0 +1,136 @@
+import { Cell, core, utils } from "@ckb-lumos/base";
+import {
+  TransactionSkeletonType,
+  createTransactionFromSkeleton,
+} from "@ckb-lumos/helpers";
+import { Reader, normalizers } from "@ckb-lumos/toolkit";
+import { Hash, Script } from "@ckb-lumos/base";
+import { BI } from "@ckb-lumos/bi";
+
+function groupInputs(inputs: Cell[], locks: Script[]): Map<string, number[]> {
+  const lockSet = new Set<string>();
+  for (const lock of locks) {
+    const scriptHash = utils
+      .ckbHash(core.SerializeScript(normalizers.NormalizeScript(lock)))
+      .serializeJson();
+    lockSet.add(scriptHash);
+  }
+
+  const groups = new Map<string, number[]>();
+  for (let i = 0; i < inputs.length; i++) {
+    const scriptHash = utils
+      .ckbHash(
+        core.SerializeScript(
+          normalizers.NormalizeScript(inputs[i].cell_output.lock)
+        )
+      )
+      .serializeJson();
+    if (lockSet.has(scriptHash)) {
+      if (groups.get(scriptHash) === undefined) groups.set(scriptHash, []);
+      groups.get(scriptHash)!.push(i);
+    }
+  }
+  return groups;
+}
+
+function calcRawTxHash(tx: TransactionSkeletonType): Reader {
+  return utils.ckbHash(
+    core.SerializeRawTransaction(
+      normalizers.NormalizeRawTransaction(createTransactionFromSkeleton(tx))
+    )
+  );
+}
+
+export interface Hasher {
+  update(message: ArrayBuffer): void;
+  digest(): Hash;
```

Comment:
`Hash` is actually an alias of a string, and we should avoid errors in Hasher's implementers, most of the hasher in the current js ecosystem use `Uint8Array` as the structure for binary data, so it is better to use a definition like this for Hasher

```ts
interface Hasher {
  update(bytes: Uint8Array): void;
  digest(): Uint8Array;
}
```

Usage:

```ts
// with NodeJS
const customHasher: Hasher = require("crypto").createHash("sha256");
```